import { TimeoutError } from '../errors/TimeoutError.js';

const timeout = async (request, abortController, options) => new Promise((resolve, reject) => {
  const timeoutId = setTimeout(() => {
    if (abortController) {
      abortController.abort();
    }

    reject(new TimeoutError(request));
  }, options.timeout);
  void options.fetch(request).then(resolve).catch(reject).then(() => {
    clearTimeout(timeoutId);
  });
});
const delay = async ms => new Promise(resolve => {
  setTimeout(resolve, ms);
});

export { delay, timeout };
